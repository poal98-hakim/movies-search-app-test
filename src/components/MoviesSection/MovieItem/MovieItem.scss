

.movie-item-wrapper{
    display: flex;
    position: relative;
    background-color: black;
    color: white;
    border-radius: 5px;
    padding: 20px;
    margin: 15px 0;
    @media only screen and (max-width:600px) {
        height: 7em;
    }
}

.left-cont{
    display: flex;
    align-items: center;
    width: 30%;
    @media only screen and (max-width:600px) {
        width: 40%;
    }
}

.right-cont{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding-left: 10px;
    width: 70%;
    @media only screen and (max-width:600px) {
        width: 60%;
    }
}



