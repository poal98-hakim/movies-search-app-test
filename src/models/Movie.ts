export interface IApiMovies{
    results: IMovie[];
    has_more: boolean;
    num_results: number;
}

export interface IMovie{
    display_title: string;
    mpaa_rating: string,
    critics_pick: number;
    byline: string,
    headline: string;
    summary_short: string;
    publication_date: string,
    opening_date: string;
    date_updated: string;
    link: ILink;
    multimedia: IMultimedia;
}

export interface ILink {
    type: string;
    url: string;
    suggested_link_text: string;
}

export interface IMultimedia {
    type: string;
    src: string;
    height: number;
    width: number;
}