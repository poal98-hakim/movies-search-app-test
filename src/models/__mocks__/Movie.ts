export const MOVIE1 = {
    display_title: "Everything Everywhere All at Once",
    mpaa_rating: "R",
    critics_pick: 1,
    byline: "A.O. Scott",
    headline: "‘Everything Everywhere All at Once’ Review: It’s Messy, and Glorious",
    summary_short: "Michelle Yeoh stars as a stressed-out laundromat owner dragged into cosmic battle and genre chaos.",
    publication_date: "2022-03-24",
    opening_date: "2022-03-25",
    date_updated: "2022-03-24 16:58:03",
    link: {
        type: "article",
        url: "https://www.nytimes.com/2022/03/24/movies/everything-everywhere-all-at-once-review.html",
        suggested_link_text: "Read the New York Times Review of Everything Everywhere All at Once"
    },
    multimedia: {
        type: "mediumThreeByTwo210",
        src: "https://static01.nyt.com/images/2022/03/24/arts/24everything-everywhere/merlin_204080367_730f3977-a2fa-425e-98db-01be8fecac21-mediumThreeByTwo440.jpg",
        height: 140,
        width: 210
    }
}
export const MOVIE2 = {
    display_title: "You Are Not My Mother",
    mpaa_rating: "R2",
    critics_pick: 1,
    byline: "Jeannette Catsoulis",
    headline: "‘You Are Not My Mother’ Review: Parental Misguidance",
    summary_short: "A lonely teenager is traumatized by her mother’s volatile behavior in this impressive horror debut.",
    publication_date: "2022-03-24",
    opening_date: "2022-03-25",
    date_updated: "2022-03-24 15:32:03",
    link: {
        type: "article",
        url: "https://www.nytimes.com/2022/03/24/movies/you-are-not-my-mother-review.html",
        suggested_link_text: "Read the New York Times Review of You Are Not My Mother"
    },
    multimedia: {
        type: "mediumThreeByTwo210",
        src: "https://static01.nyt.com/images/2022/03/25/arts/mother1/mother1-mediumThreeByTwo440.jpg",
        height: 140,
        width: 210
    }
}

export const API_MOVIES_RESULT = {
    results: [MOVIE1, MOVIE2],
    num_results: 2,
    hasMore: true
}