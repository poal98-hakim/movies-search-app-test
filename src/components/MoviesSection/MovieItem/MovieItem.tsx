import React from "react";
import { IMovie } from "../../../models/Movie";
import Poster from "./Poster/Poster";
import ContentMovie from "./ContentMovie/ContentMovie";
import "./MovieItem.scss";

interface IProps{
  movie: IMovie;
}

const MovieItem : React.VFC<IProps> = ({ movie }) => {
  
  const {multimedia} = movie;

  return (
    <article className="movie-item-wrapper">
      <div className="left-cont">
        <Poster multimedia={multimedia} />
      </div>
      <div className="right-cont">
        <ContentMovie movie={movie} />
      </div>
    </article>
  )
}
export default MovieItem;

