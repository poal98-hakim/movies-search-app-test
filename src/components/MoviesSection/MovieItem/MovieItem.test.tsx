import MovieItem from "./MovieItem";
import { render, screen } from '@testing-library/react';
import { API_MOVIES_RESULT } from "../../../models/__mocks__/Movie"
import moment from 'moment';

const PROPS = {
  movie: API_MOVIES_RESULT.results[0]
}

describe('MovieItem', () => {
  let props = {...PROPS};//using a shallow copy because I'm only reading the data. Otherwise due to the nested array I would do a deep copy

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<MovieItem {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<MovieItem {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads poster image', () => {
    expect(screen.getByAltText("poster")).toBeInTheDocument();
    expect(screen.getByAltText("poster")).toHaveAttribute("src", props.movie.multimedia.src);
  })

  it('loads title movie', () => {
    expect(screen.getByText(props.movie.display_title)).toBeInTheDocument();
  })

  it('loads movie description', () => {
    expect(screen.getByText(props.movie.summary_short)).toBeInTheDocument();
  })

  it('loads publication date', () => {
    const dateFormatted = moment(props.movie.publication_date).format('LL');
    expect(screen.getByText(dateFormatted)).toBeInTheDocument();
  })
  
})
