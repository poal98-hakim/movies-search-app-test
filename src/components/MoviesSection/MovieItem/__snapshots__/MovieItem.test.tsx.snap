// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem renders correctly 1`] = `
<body>
  <div>
    <article
      class="movie-item-wrapper"
    >
      <div
        class="left-cont"
      >
        <img
          alt="poster"
          class="poster full-width"
          src="https://static01.nyt.com/images/2022/03/24/arts/24everything-everywhere/merlin_204080367_730f3977-a2fa-425e-98db-01be8fecac21-mediumThreeByTwo440.jpg"
        />
      </div>
      <div
        class="right-cont"
      >
        <section
          class="flex f-direction-column f-justify-between full-height"
        >
          <div>
            <header>
              <h1
                class="movie-title"
              >
                Everything Everywhere All at Once
              </h1>
            </header>
            <a
              class="subtitle"
              href="https://www.nytimes.com/2022/03/24/movies/everything-everywhere-all-at-once-review.html"
              rel="noreferrer"
              target="_blank"
            >
              Read the New York Times Review of Everything Everywhere All at Once
            </a>
            <p
              class="movie-description"
            >
              Michelle Yeoh stars as a stressed-out laundromat owner dragged into cosmic battle and genre chaos.
            </p>
          </div>
          <footer>
            <span
              class="date-movie"
              data-testid="date-movie"
            >
              March 24, 2022
            </span>
          </footer>
        </section>
      </div>
    </article>
  </div>
  <div>
    <article
      class="movie-item-wrapper"
    >
      <div
        class="left-cont"
      >
        <img
          alt="poster"
          class="poster full-width"
          src="https://static01.nyt.com/images/2022/03/24/arts/24everything-everywhere/merlin_204080367_730f3977-a2fa-425e-98db-01be8fecac21-mediumThreeByTwo440.jpg"
        />
      </div>
      <div
        class="right-cont"
      >
        <section
          class="flex f-direction-column f-justify-between full-height"
        >
          <div>
            <header>
              <h1
                class="movie-title"
              >
                Everything Everywhere All at Once
              </h1>
            </header>
            <a
              class="subtitle"
              href="https://www.nytimes.com/2022/03/24/movies/everything-everywhere-all-at-once-review.html"
              rel="noreferrer"
              target="_blank"
            >
              Read the New York Times Review of Everything Everywhere All at Once
            </a>
            <p
              class="movie-description"
            >
              Michelle Yeoh stars as a stressed-out laundromat owner dragged into cosmic battle and genre chaos.
            </p>
          </div>
          <footer>
            <span
              class="date-movie"
              data-testid="date-movie"
            >
              March 24, 2022
            </span>
          </footer>
        </section>
      </div>
    </article>
  </div>
</body>
`;
